Makefile

```diff
@@ -518,6 +518,7 @@ $(eval $(call assert_boolean,USE_COHERENT_MEM))
 $(eval $(call assert_boolean,USE_TBBR_DEFS))
 $(eval $(call assert_boolean,WARMBOOT_ENABLE_DCACHE_EARLY))
 $(eval $(call assert_boolean,BL2_AT_EL3))
+$(eval $(call assert_boolean,BL2_AT_EL3_ROM))
```

Comment:
I think the new flag should be called BL2_IN_XIP_MEM. The term ROM is confusing in this context. Also, this feature potentially could be used when BL2 is in Secure-EL1, so this should be orthgonal to the BL2_AT_EL3 feature. Having said that, for now you could avoid supporting the case where BL2_AT_EL3 = 0 and BL2_IN_XIP_MEM = 1, by throwing an error in the top level Makefile (in the "Check incompatible options" section).

Given the above, I think your commit message should look something like this:

Add support for BL2 in XIP memory

In some use-cases BL2 will be stored in eXecute In Place (XIP) memory, like BL1. In these use-cases, it is necessary to initialize the RW sections in RAM, while leaving the RO sections in place. This patch enable this use-case with a new build option, BL2_IN_XIP_MEM. For now, this option is only supported when BL2_AT_EL3 is 1.

Reply:
Great, will modify it, thanks.